### PR TITLE
Chore: clean up .gitignore

### PR DESCRIPTION
**Pull request by camdencheek**
Just a quick cleanup while I was taking a look at the .gitignore file to figure out why some files were unintentionally being ignored.

## Test plan

Removing these did not uncover any ignored files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -177,8 +164,8 @@ index.scip
 .aspect/bazelrc/ci.generated.bazelrc
 
 # nix
-result
-result-*
+/result
+/result-*
```
Comment (camdencheek) [inline .gitignore:170]: @Strum355, this was causing the `internal/search/result` directory to be ignored. Assuming that wasn't intentional, I guessed that any `nix` `result` directories will be at the root of the repo. Is that right?

File: .gitignore [hunk 2]
```diff
@@ -132,10 +123,6 @@ sg.config.overwrite.yaml
 # sg Google Cloud API OAuth token
 .sg.token.json
 
-# Generated sitemaps are not committed, they're hosted in a GCS bucket.
-sitemap/
-sitemap_query.db
-
```
Comment (camdencheek) [inline .gitignore:138]: Sitemap tool no longer exists

File: .gitignore [hunk 3]
```diff
@@ -101,9 +95,6 @@ graphql-operations.ts
 *.module.css.d.ts
 dll-bundle
 
-# Extensions
-/client/extension-api/dist
-
```
Comment (camdencheek) [inline .gitignore:106]: Extension API build no longer goes to the `dist` dir

File: .gitignore [hunk 4]
```diff
@@ -78,11 +78,7 @@ eb-bundle.zip
 
 .gtm/
 
-cmd/src/debug
 cmd/gitserver/debug
-cmd/indexer/debug
-
-/.gtm/
```
Comment (camdencheek) [inline .gitignore:85]: Duplicated by a more general rule above.

File: .gitignore [hunk 5]
```diff
@@ -78,11 +78,7 @@ eb-bundle.zip
 
 .gtm/
 
-cmd/src/debug
 cmd/gitserver/debug
-cmd/indexer/debug
```
Comment (camdencheek) [inline .gitignore:83]: No longer exist

File: .gitignore (hunk 1 above)
Comment (camdencheek) [reply .gitignore:170]: cc @jasonhawkharris -- this is why you couldn't find `internal/search/result/match.go` in your editor that one time

**Review by eseliger — APPROVED** [review]
Thanks for cleanup :)

File: .gitignore (hunk 1 above)
Comment (Strum355) [reply .gitignore:170]: Thats correct yea, thx for the fix!

**Pull request merged**